src/Runner.Worker/Container/ContainerHooks/ContainerHookManager.cs

```diff
@@ -0,0 +1,259 @@
+                    PrependPath = prependPath,
+                    WorkingDirectory = workingDirectory,
+                },
+                State = GetHookStateInJson(context)
+            };
+
+            var response = await ExecuteHookScript<HookResponse>(context, input, ActionRunStage.Main, prependPath);
+            if (response == null)
+            {
+                return;
+            }
+            SaveHookState(context, response.State, input);
+        }
+
+        private async Task<T> ExecuteHookScript<T>(IExecutionContext context, HookInput input, ActionRunStage stage, string prependPath) where T: HookResponse
+        {
+            var scriptDirectory = Path.GetDirectoryName(HookIndexPath);
+            var stepHost = HostContext.CreateService<IDefaultStepHost>();
+            Dictionary<string, string> inputs = new()
+            {
+                ["standardInInput"] = JsonUtility.ToString(input),
+                ["path"] = HookIndexPath,
+                ["shell"] = HostContext.GetDefaultShellForScript(HookIndexPath, Trace, prependPath)
+            };
+
+            var handlerFactory = HostContext.GetService<IHandlerFactory>();
+            var handler = handlerFactory.Create(
+                            context,
+                            null,
+                            stepHost,
+                            new ScriptActionExecutionData(),
+                            inputs,
+                            environment: new Dictionary<string, string>(VarUtil.EnvironmentVariableKeyComparer),
+                            context.Global.Variables,
+                            actionDirectory: scriptDirectory,
+                            localActionContainerSetupSteps: null) as ScriptHandler;
+            handler.PrepareExecution(stage);
+
+            IOUtil.CreateEmptyFile(input.ResponseFile);
+            await handler.RunAsync(stage);
+            if (handler.ExecutionContext.Result == TaskResult.Failed)
+            {
+                throw new Exception($"The hook script at '{HookIndexPath}' running command '{input.Command}' did not execute successfully."); // TODO: better exception
+            }
+            var response = GetResponse<T>(input);
+            return response;
+        }
+
+        private string GenerateResponsePath()
+        {
+            return Path.Combine(HostContext.GetDirectory(WellKnownDirectory.Temp), ResponseFolderName, $"{Guid.NewGuid()}.json");
+        }
+
+        private string GetPrependPath(IExecutionContext context)
+        {
+            return string.Join(Path.PathSeparator.ToString(), context.Global.PrependPath.Reverse<string>()); ;
+        }
+
+        private T GetResponse<T>(HookInput input) where T: HookResponse
+        {
+            T response = null;
+
+            if (!string.IsNullOrEmpty(input.ResponseFile) && File.Exists(input.ResponseFile))
+            {
+                response = IOUtil.LoadObject<T>(input.ResponseFile);
```

Comment:
we should validate this throws if they write xml or something

Reply:
Currently we are validating if `IsAlpine` is null. Should we validate the rest of the properties?